pkg/systemstatsmonitor/net_collector_test.go

```diff
@@ -0,0 +1,189 @@
+func testCollectAux(t *testing.T, name string, excludeInterfaceRegexp ssmtypes.NetStatsInterfaceRegexp, validate func(*testing.T, *netCollector)) {
+	// mkdir /tmp/proc-X
+	procDir, err := ioutil.TempDir(os.TempDir(), "proc-")
+	if err != nil {
+		t.Fatalf("Failed to create temp proc directory: %v", err)
+	}
+	// rm -r /tmp/proc-X
+	defer os.RemoveAll(procDir)
+	// mkdir -C /tmp/proc-X/net
+	procNetDir := path.Join(procDir, "net")
+	if err := os.Mkdir(procNetDir, 0777); err != nil {
+		t.Fatalf("Failed to create directory %q: %v", procNetDir, err)
+	}
+
+	// touch /tmp/proc-X/net/dev
+	filename := path.Join(procNetDir, "dev")
+	f, err := os.Create(filename)
+	if err != nil {
+		t.Fatalf("Failed to create file %q: %v", filename, err)
+	}
+	// echo $FILE_CONTENT > /tmp/proc-X/net/dev
+	if _, err = f.WriteString(fakeNetProcContent); err != nil {
+		t.Fatalf("Failed to write to file %q: %v", filename, err)
+	}
+	if err = f.Close(); err != nil {
+		t.Fatalf("Failed to close file %q: %v", filename, err)
+	}
+
+	// Build the netCollector
+	config := &ssmtypes.NetStatsConfig{
+		ExcludeInterfaceRegexp: excludeInterfaceRegexp,
+		MetricsConfigs:         defaultMetricsConfig,
+	}
+	netCollector := &netCollector{
+		config:   config,
+		procPath: procDir,
+		recorder: newIfaceStatRecorder(newFakeInt64Metric),
+	}
+	netCollector.initOrDie()
+	netCollector.collect()
+	validate(t, netCollector)
+}
+
+func TestCollect(t *testing.T) {
+	tcs := []struct {
+		Name                   string
+		ExcludeInterfaceRegexp ssmtypes.NetStatsInterfaceRegexp
+		Validate               func(t *testing.T, nc *netCollector)
+	}{
+		{
+			Name:                   "NoFilterMatch",
+			ExcludeInterfaceRegexp: ssmtypes.NetStatsInterfaceRegexp{R: regexp.MustCompile(`^fake$`)},
+			Validate: func(t *testing.T, nc *netCollector) {
+				// We just validate two metrics, no need to check all of them
+				expectedValues := map[metrics.MetricID]map[string]int64{
+					metrics.NetDevRxBytes: map[string]int64{
+						"eth0":    5000,
+						"docker0": 1000,
+						"docker1": 500,
+						"docker2": 0,
+					},
+					metrics.NetDevTxBytes: map[string]int64{
+						"eth0":    2500,
+						"docker0": 0,
+						"docker1": 3000,
+						"docker2": 6000,
+					},
+				}
+				for metricID, interfaceValues := range expectedValues {
+					collector, ok := nc.recorder.collectors[metricID]
+					if !ok {
+						t.Errorf("Failed to get collector of metric %s", metricID)
+						continue
+					}
+					fakeInt64Metric, ok := collector.metric.(*metrics.FakeInt64Metric)
+					if !ok {
+						t.Fatalf("Failed to convert metric %s to fakeMetric", string(metricID))
+					}
+					for _, repr := range fakeInt64Metric.ListMetrics() {
+						interfaceName, ok := repr.Labels[interfaceNameLabel]
+						if !ok {
+							t.Fatalf("Failed to get label %q for ", interfaceNameLabel)
+						}
+						expectedValue, ok := interfaceValues[interfaceName]
+						if !ok {
+
```

Comment:
same as below, is it intended for this case to be empty?

Reply:
Fixed.